### PR TITLE
feat:Limit SQL length to avoid Stack Overflow

### PR DESCRIPTION
**Pull request by yanni97**
1. The catch exception type is changed to Throwable
2. Limit SQL length to avoid Stack Overflow
3. The maximum length of sql is read from the configuration
4. Replace the [ and ] symbols

### CODE REVIEW

File: arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java [hunk 1]
```diff
@@ -25,14 +30,18 @@
  * @since 1.0.0
  */
 @Slf4j
+@Component
 public class DatabaseUtils {
```
Comment (coryhh) [inline arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java:34]: modify the name of class and move the location of the class

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/config/ConfigManager.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import javax.annotation.Resource;
+import java.util.Map;
+
+/**
+ * @author niyan
+ * @date 2024/6/6
+ * @since 1.0.0
+ */
+@Component
+public class ConfigManager {
```
Comment (coryhh) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/config/ConfigManager.java:17]: Consider the implementation of conditional replacement

File: arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java [hunk 3]
```diff
@@ -121,10 +130,18 @@ public static List<String> parseTableNames(String operationName) {
      * @return table schema info
      */
     @VisibleForTesting
-    static TableSchema parse(String sql) {
+    public TableSchema parse(String sql) {
         if (StringUtils.isEmpty(sql)) {
             return null;
         }
+        String maxSqlLength = configManager.getConfig("maxSqlLength");
+        int maxSqlLengthInt = StringUtils.isEmpty(maxSqlLength) ? 50000 : Integer.parseInt(maxSqlLength);
```
Comment (lucas-myx) [inline arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java:138]: int maxSqlLengthInt = configManager.getInt("maxSqlLength", 50000);

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/config/ConfigManager.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Resource
+    private ConfigProvider configProvider;
+
+    @PostConstruct
+    public void init() {
+        configProvider.loadConfigs("/META-INF/app.properties");
```
Comment (pangdayuan1) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/config/ConfigManager.java:24]: Should the "application.yaml" configuration file be loaded?

File: arex-storage-web-api/src/test/java/com/arextest/storage/utils/DatabaseUtilsTest.java [hunk 5]
```diff
@@ -170,4 +173,18 @@ public void parseSQLWithKeyWord() {
...
+    @Test
+    public void parseSQLWithSquareBrackets() {
+        String sql = "SELECT * FROM [flight_transaction_sub] WITH (NOLOCK) WHERE [orderid]=?";
+        TableSchema tableSchema = DatabaseUtils.parse(sql);
+        assertEquals("flight_transaction_sub", StringUtils.join(tableSchema.getTableNames(), ","));
+    }
+
+    @Test
+    public void parseSQLWithAnnotationSymbols() {
+        String sql = "/*appid:100026335 author:meiyu_liu reviewcode:cUvxLC202312210117 */ SELECT count(DISTINCT o.orderId) FROM orderbasicinfo AS o JOIN orderpassengerinfo AS p ON o.orderId = p.orderId WHERE -- 审批人uid 【必传】 o.uid IN ( ? ) AND /* corpId */ o.corpId = ? AND -- 产线 o.productLine IN ( ? , ? , ? , ? ) AND /* 时间限制范围 */ ( /* 预定时间、出行时间、出行UTC时间 【必传其一】*/ o.orderDate BETWEEN ? AND ? ) AND /* 我的订单过滤条件 */ ( /* 订单状态*/ /* 审批状态*/ o.approvalStatus IN ( ? ) /* 是否出行人*/ ) AND o.orderStatus != 'UnSubmit' /* 默认过滤已删除订单 */ AND ifnull(o.delete_flag, false) = false limit 3000";
```
Comment (QizhengMo) [inline arex-storage-web-api/src/test/java/com/arextest/storage/utils/DatabaseUtilsTest.java:186]: please remove sensitive data from test case

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/config/ApplicationDefaultConfig.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * @author niyan
+ * @date 2024/6/6
+ * @since 1.0.0
+ */
+@Component
+@Slf4j
+public class ApplicationDefaultConfig {
+
+    @Resource(name = "applicationPropertiesConfigProvider")
```
Comment (coryhh) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/config/ApplicationDefaultConfig.java:19]: no need to specify bean name

**Review by coryhh — DISMISSED** [review]

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/config/impl/ApplicationPropertiesConfigProvider.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        return environment.getProperty(key, String.class, StringUtils.EMPTY);
+    }
+
+    @Override
+    public int getConfigAsInt(String key, int defaultValue) {
+        try {
+            return environment.getProperty(key, Integer.class, defaultValue);
+        } catch (Exception e) {
+            return defaultValue;
+        }
```
Comment (mr3) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/config/impl/ApplicationPropertiesConfigProvider.java:44]: Move this method to `ApplicationDefaultConfig`

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/config/provider/ConfigProvider.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+    /**
+     * get config as string
+     */
+    String getConfigAsString(String key);
+
+    /**
+     * get config as int
+     */
+    int getConfigAsInt(String key, int defaultValue);
```
Comment (mr3) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/config/provider/ConfigProvider.java:29]: Remove this method.

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/config/ApplicationDefaultConfig.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ *
+ * @author niyan
+ * @date 2024/6/6
+ * @since 1.0.0
+ */
+@Component
+@Slf4j
+public class ApplicationDefaultConfig {
```
Comment (mr3) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/config/ApplicationDefaultConfig.java:17]:
```java
AbstractConfig {

protected String getConfigAsString(String key);

public String getConfigAsString(String key, String default) {
}

public String getConfigAsInt(String key, String default) {
}
```

public class ApplicationDefaultConfig extend AbstractConfig;

**Review by coryhh — APPROVED** [review]

**Review by mr3 — APPROVED** [review]

File: arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java [hunk 10]
```diff
@@ -30,9 +36,9 @@ public class DatabaseUtils {
     private DatabaseUtils() {
```
Comment (coryhh) [inline arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java:36]: what is the purpose of it

Comment (mr3) [reply arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java:36]: sonarqube rule: Hide Utility Class Constructor

**Review by mr3 — APPROVED** [review]

**Review by coryhh — APPROVED** [review]

**Pull request merged**